MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -57,30 +67,52 @@ protected virtual void Reset()
     public class MvxAppStart<TViewModel> : MvxAppStart
         where TViewModel : IMvxViewModel
     {
-        protected readonly IMvxNavigationService NavigationService;
-
-        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application)
+        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application, navigationService)
         {
-            NavigationService = navigationService;
         }
 
-        protected override void Startup(object hint = null)
+        protected override void NavigateToFirstViewModel(object hint)
         {
-            base.Startup(hint);
+            try
+            {
+                NavigationService.Navigate<TViewModel>().GetAwaiter().GetResult();
+            }
+            catch (System.Exception exception)
+            {
+                throw exception.MvxWrap("Problem navigating to ViewModel {0}", typeof(TViewModel).Name);
+            }
+        }
+    }
 
-            NavigateToFirstViewModel(hint);
+    public class MvxAppStart<TViewModel, TParameter> : MvxAppStart<TViewModel> where TViewModel : IMvxViewModel<TParameter>
+    {
+        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application, navigationService)
+        {
         }
 
-        protected virtual void NavigateToFirstViewModel(object hint)
+        protected override object ApplicationStartup(object hint = null)
         {
-            if (hint != null)
+            var applicationHint = base.ApplicationStartup(hint);
+            if (applicationHint is TParameter parameter && Application is IMvxApplication<TParameter> typedApplication)
             {
-                MvxLog.Instance.Trace("Hint ignored in default MvxAppStart");
+                //There is no way to pass the hint back
+                return typedApplication.Startup(parameter);
```

Comment:
Given this issue, perhaps we should consider passing around a new type, IStartupHint (and associated IStartupHint<THint>) that has a property Hint. Rather than passing back an updated hint, the Hint within the IStartupHint object could be updated.

Reply:
This was a leftover comment, not an issue anymore since breaking the interface method.